### PR TITLE
Closes #1576 1576_update_param_tte: keep only obs for subjects in dataset_adsl

### LINKED ISSUE #1576: Feature Request: `derive_param_tte()` should not create observations for subjects not in ADSL
### Feature Idea

Currently `derive_param_tte()` creates observations for subjects which have an event or are censored but who are not in ADSL. For those `STARTDT` is `NA`, which implies that `AVAL` is `NA` as well. So these observations are useless for time-to-event analyses.

`derive_param_tte()` should be updated such that such observations are not created.

Additionally the documentation should be updated where it mentions the `start_imputation_flag` argument. This argument does not exist.

### Relevant Input

_No response_

### Relevant Output

_No response_

### Reproducible Example/Pseudo Code

_No response_

### PR DESCRIPTION
**Pull request by bundfussr**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiraldev/devel/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/devel/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#deprecation)?
- [x] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately 
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

File: R/derive_param_tte.R [hunk 1]
```diff
@@ -111,13 +111,15 @@
...
-#'   1. the variables specified for `start_date` and `start_imputation_flag` are
-#'   joined from the ADSL dataset,
-#'   1. the variables as defined by the `set_values_to` parameter are added,
-#'   1. the `ADT`/`ADTM` variable is set to the maximum of `ADT`/`ADTM` and
-#'   `STARTDT`/`STARTDTM` (depending on the `create_datetime` parameter), and
-#'   1. the new observations are added to the output dataset.
+#'   Finally:
+#'   1. The variables specified for `start_date` and `start_imputation_flag` are
+#'   joined from the ADSL dataset. Only subjects in both datasets are kept,
+#'   i.e., subjects with and event or censoring and an observation in
```
Comment (sadchla-codes) [inline R/derive_param_tte.R:117]: minor typo here as well, but is this statement intend to read "i.e., subjects **with an event** or censoring and an observation in" rather than "i.e., subjects** with and event** or censoring and an observation in"?

**Review by sadchla-codes — CHANGES_REQUESTED** [review]

**Review by sadchla-codes — APPROVED** [review]

File: NEWS.md [hunk 2]
```diff
@@ -1,3 +1,17 @@
+# admiral 1.0.0
+
+## New Features
+
+## Updates of Existing Functions
+
+## Breaking Changes
+
+- Function `derive_param_tte()` has been updated such that only observations are
+added for subjects who have both an event or censoring and and an observation in
```
[the remark this answers is not in the document]
Comment (bundfussr) [reply NEWS.md]: Thanks, fixed.

File: R/derive_param_tte.R (hunk 1 above)
Comment (bundfussr) [reply R/derive_param_tte.R]: Yes, this is ambiguous. I meant "(an event or censoring) and an observation in `dataset_adsl`".

Is the new version clearer?

Comment (sadchla-codes) [reply R/derive_param_tte.R]: Yes, this new version is better. Thank you.

**Comment by bundfussr** [comment]
@sadchla-codes , the "Check Templates" workflow did not start.

Could you reapprove? Maybe this triggers the workflow.

**Review by sadchla-codes — APPROVED** [review]

**Comment by sadchla-codes** [comment]
@bundfussr reapproved! it looks like it's going now
